client/src/hooks/helpers/useArmies.tsx

```diff
@@ -297,30 +294,32 @@ export const getArmiesByBattleId = (battle_id: bigint) => {
         Protectee,
         EntityName,
         Stamina,
-        Battle,
       },
     },
     account: { account },
   } = useDojo();
 
-  const armiesEntityIds = runQuery([HasValue(Army, { battle_id })]);
-  return formatArmies(
-    Array.from(armiesEntityIds),
-    account.address,
-    Army,
-    Protectee,
-    EntityName,
-    Health,
-    Quantity,
-    Movable,
-    Capacity,
-    ArrivalTime,
-    Position,
-    EntityOwner,
-    Owner,
-    Realm,
-    Stamina,
-  ).filter((army) => isArmyAlive(army, Battle, Army));
+  const armiesByBattleId = (battle_id: bigint) => {
```

Comment:
Why wrap the call ? Cant we pass the battle_id directly to the getArmiesByBattleId ?

Reply:
I'm calling this function inside a useMemo so the call of `useDojo` would make it error out. I'm also using some specific filtering inside this `useMemo`, so to keep it generic I prefer not doing that filtering inside the `getArmiesByBattleId` function